src/libsyncengine/syncpal/isyncworker.h

```diff
@@ -32,7 +32,7 @@ namespace KDC {
 
 class ISyncWorker {
     public:
-        ISyncWorker(std::shared_ptr<SyncPal> syncPal, const std::string &name, const std::string &shortName);
+        ISyncWorker(std::shared_ptr<SyncPal> syncPal, const std::string &name, const std::string &shortName, bool testing = false);
```

Comment:
Using a derived class for unit tests would enable us not to use test-related variables in deployed code.
Besides, it would offer more flexibility than a single flag.  To be discussed.

Reply:
Agree. If ok for you, let's do it in anpther dedicated issue (https://infomaniak.atlassian.net/browse/KDESKTOP-931 for example)